src/sys/wasi/net.rs

```diff
@@ -0,0 +1,45 @@
+#![allow(dead_code)]
+
+use std::io;
+
+/// A lot of function are not support on Wasi, this function returns a
+/// consistent error when calling those functions.
+#[inline]
+fn unsupported() -> io::Error {
+    io::Error::new(io::ErrorKind::Other, "not supported on wasi")
+}
+
+pub(crate) mod tcp {
+    use std::io;
+    use std::net;
+
+    use super::unsupported;
+
+    pub type TcpSocket = wasi::Fd;
+
+    pub(crate) fn new_for_addr(_address: net::SocketAddr) -> io::Result<i32> {
+        Err(unsupported())
+    }
+
+    pub(crate) fn bind(_: &net::TcpListener, _: net::SocketAddr) -> io::Result<()> {
+        Err(unsupported())
+    }
+
+    pub(crate) fn connect(_: &net::TcpStream, _: net::SocketAddr) -> io::Result<net::TcpStream> {
+        Err(unsupported())
+    }
+
+    pub(crate) fn listen(_: &net::TcpListener, _: u32) -> io::Result<()> {
+        Err(unsupported())
+    }
+
+    pub(crate) fn set_reuseaddr(_: &net::TcpListener, _: bool) -> io::Result<()> {
+        Err(unsupported())
+    }
```

Comment:
It is true that this is unsupported. However, I think we can also safely lie here and just return `Ok(())`. I don't see any negative consequences of this.

Reply:
sure, we can lie... but it does not gain anything for the mio internals.